### PR TITLE
[cli][outputs] the cli will handle instances of the secrets bucket not existing better

### PR DESCRIPTION
**Pull request by ryandeivert**
to @jacknagz 
cc @airbnb/streamalert-maintainers 
size: small

## changes ##
* The cli will handle instances of the secrets bucket not existing better.
* Also updating docs to include a note about terraforming on first run to get the `streamalert.secrets` bucket.

### CODE REVIEW

File: docs/source/outputs.rst [hunk 1]
```diff
@@ -34,6 +34,10 @@ Adding a new configuration for a currently supported service is handled using ``
 For example:
  - ``python stream_alert_cli.py output new --service slack``
 
+.. Note:: If this is the first time you have configured new outputs via the cli, you may see this error for certain services:
+ `An error occurred while sending credentials to S3 for key [<SERVICE>/<KEY>]: The specified bucket does not exist [<PREFIX>.streamalert.secrets]`
+ If you encounter this error, you must first run `python stream_alert_cli.py terraform build`. This ensures the S3 bucket used for storing encrypted secrets is created and only needs to occur once.
```
Comment (mime-frame) [inline docs/source/outputs.rst]: and only needs to be run once.

Comment (ryandeivert) [inline docs/source/outputs.rst]: funny because I was tossing these two verbiages back and forth and couldn't settle on one

**Comment by ryandeivert** [comment]
@mime-frame PTAL - tweaked a bit. also removed old `secrets.rst` file

File: docs/source/secrets.rst [hunk 2]
```diff
@@ -1,33 +0,0 @@
-Secrets
```
Comment (mime-frame) [inline docs/source/secrets.rst]: @ryandeivert why are we deleting the entire `secrets` doc and the link to it?

Comment (ryandeivert) [inline docs/source/secrets.rst]: This page described the old way of storing secret info, and should have been removed when the outputs refactor happened but was overlooked.

We no longer require the user to use `aws kms` from the command line and handle all the overhead for them.

Comment (mime-frame) [inline docs/source/secrets.rst]: @ryandeivert i might be derp'ing, but `docs/source/outputs.rst` doesn't appear to cover the new way of doing secrets, where is it covered?

Comment (ryandeivert) [inline docs/source/secrets.rst]: @mime-frame Just pushed a commit to also remove the link from the TOC in `index.rst`

Comment (ryandeivert) [inline docs/source/secrets.rst]: @mime-frame secrets are transparently sent kms/s3 behind the scenes and the process outlined `outputs.html#configuration` walks the user through adding the output.

As simple as:
`python stream_alert_cli.py output new --service slack`

Which then prompts for things like channel name and integration key (the latter of which is sent to kms/s3 for encryption/storage).

**Pull request merged**